### PR TITLE
Vert.x restructuring - part 1/2

### PR DESCRIPTION
**Pull request by mkouba**
- merge the quarkus-vertx and quarkus-vertx-core extentions
- keep the API packages, i.e. io.quarkus.vertx.ConsumeEvent and
io.quarkus.vertx.LocalEventBusCodec
- the final extension name is "quarkus-vertx"
- see also #16797

### CODE REVIEW

File: extensions/pom.xml [hunk 1]
```diff
@@ -30,7 +30,6 @@
         <module>jsonb</module>
 
         <!-- HTTP -->
-        <module>vertx-core</module>
```
Comment (cescoffier) [inline extensions/pom.xml:33]: I would have moved the vertx module here.

File: extensions/vertx/deployment/pom.xml [hunk 2]
```diff
@@ -61,11 +53,6 @@
             <artifactId>rest-assured</artifactId>
             <scope>test</scope>
         </dependency>
-        <dependency>
-            <groupId>io.quarkus</groupId>
```
Comment (cescoffier) [inline extensions/vertx/deployment/pom.xml:65]: Wasn't that one used?

**Review by cescoffier — COMMENTED** [review]
Thanks for working on this! That would simplify a lot!

It looks like some CI issues are related.

File: extensions/pom.xml (hunk 1 above)
Comment (mkouba) [reply extensions/pom.xml:33]: Hm, it's under the `<!-- Reactive -->` group which is probably more appropriate? The `vertx-web` module (which I plan to remove in the second part of PR) could be moved under the `<!-- HTTP -->` group though.

File: extensions/vertx/deployment/pom.xml (hunk 2 above)
Comment (mkouba) [reply extensions/vertx/deployment/pom.xml:65]: We had to move the related tests elsewhere because of a cyclic dependency...

Comment (mkouba) [reply extensions/vertx/deployment/pom.xml:65]: And in fact, it was the only one ;-)

**Comment by mkouba** [comment]
@radcortez MP Config TCKs failures - does it ring a bell?

**Comment by radcortez** [comment]
> @radcortez MP Config TCKs failures - does it ring a bell?

It is fishy that all of them are related with the MP `@ConfigProperties` tests. I'm wondering if any of these changes broke the registration somehow. 

That is done here: https://github.com/quarkusio/quarkus/blob/main/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ConfigBuildStep.java#L414-L419

**Comment by mkouba** [comment]
> I'm wondering if any of these changes broke the registration somehow.

Which changes do you mean?

**Comment by radcortez** [comment]
> > I'm wondering if any of these changes broke the registration somehow.
> 
> Which changes do you mean?

Was not referring to anything in particular, but for some reason, it seems that affected the `@ConfigProperties` somehow. Let me checkout your PR and try to figure out what is going on.

**Comment by mkouba** [comment]
> > > I'm wondering if any of these changes broke the registration somehow.
> > 
> > 
> > Which changes do you mean?
> 
> Was not referring to anything in particular, but for some reason, it seems that affected the `@ConfigProperties` somehow. Let me checkout your PR and try to figure out what is going on.

FYI I wasn't able to reproduce the problem locally...

**Comment by radcortez** [comment]
> > > > I'm wondering if any of these changes broke the registration somehow.
> > > 
> > > 
> > > Which changes do you mean?
> > 
> > 
> > Was not referring to anything in particular, but for some reason, it seems that affected the `@ConfigProperties` somehow. Let me checkout your PR and try to figure out what is going on.
> 
> FYI I wasn't able to reproduce the problem locally...

Same. Maybe it is some kind of CI hiccup. Lets restart the build.

**Comment by mkouba** [comment]
The CI failure does not seem to be related... Let's try again :-(

**Review by cescoffier — APPROVED** [review]
Thanks!

**Pull request merged**